chains/solana/contracts/tests/ccip/tokenpool_test.go

```diff
@@ -454,6 +454,54 @@ func TestTokenPool(t *testing.T) {
 							}
 						})
 
+						t.Run("Rate Limit Admin", func(t *testing.T) {
+							// set invalid a new rate limit admin
+							ixRateAdmin, err := test_token_pool.NewSetRateLimitAdminInstruction(
+								p.Mint,
+								user.PublicKey(),
+								poolConfig,
+								anotherAdmin.PublicKey(),
+							).ValidateAndBuild()
+							require.NoError(t, err)
+
+							// test new rate limit admin
+							ixRatesValid, err := test_token_pool.NewSetChainRateLimitInstruction(config.EvmChainSelector, p.Mint,
+								test_token_pool.RateLimitConfig{
+									Enabled:  true,
+									Capacity: amount,
+									Rate:     1,
+								}, test_token_pool.RateLimitConfig{
+									Enabled:  false,
+									Capacity: 0,
+									Rate:     0,
+								}, poolConfig, p.Chain[config.EvmChainSelector], user.PublicKey(), solana.SystemProgramID).ValidateAndBuild()
+							require.NoError(t, err)
+
+							// undo rate limit admin
+							ixRateAdmin2, err := test_token_pool.NewSetRateLimitAdminInstruction(
+								p.Mint,
+								anotherAdmin.PublicKey(),
+								poolConfig,
+								anotherAdmin.PublicKey(),
+							).ValidateAndBuild()
+							require.NoError(t, err)
+
+							// try to modify rate limit with invalid admin
+							ixRates, err := test_token_pool.NewSetChainRateLimitInstruction(config.EvmChainSelector, p.Mint,
+								test_token_pool.RateLimitConfig{
+									Enabled:  true,
+									Capacity: amount,
+									Rate:     1,
+								}, test_token_pool.RateLimitConfig{
+									Enabled:  false,
+									Capacity: 0,
+									Rate:     0,
+								}, poolConfig, p.Chain[config.EvmChainSelector], user.PublicKey(), solana.SystemProgramID).ValidateAndBuild()
+							require.NoError(t, err)
+
+							testutils.SendAndFailWith(ctx, t, solanaGoClient, []solana.Instruction{ixRateAdmin, ixRatesValid, ixRateAdmin2, ixRates}, user, config.DefaultCommitment, []string{"SetChainRateLimit"}, common.AddSigners(anotherAdmin))
```

Comment:
That `[]string{"SetChainRateLimit"}` only checks that that string is present in the logs, meaning that it will only check that the method was invoked. It won't enforce that that is the method that's failing, nor which of the two invocations you're making to is the one that fails.

I'll approve the PR anyway as I think this is minor compared to other priorities, but if you have the time to improve this check, please do so 🙂. You could check the events to make sure that the first call to that method did work for example.